### PR TITLE
security part 1 done

### PR DESCRIPTION
**Pull request by Rommelua**

### CODE REVIEW

**Review by aleksandr-hrankin — APPROVED** [review]

**Comment by aleksandr-hrankin** [comment]
Good job!

File: src/main/java/com/internet/shop/controller/LoginController.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        String password = req.getParameter("psw");
+        try {
+            User user = authService.login(login, password);
+            HttpSession session = req.getSession();
+            session.setAttribute(USER_ID, user.getId());
+            resp.sendRedirect(req.getContextPath() + "/");
+        } catch (AuthenticationException e) {
+            String errorMsg = e.getMessage();
+            req.setAttribute("errorMsg", errorMsg);
+            req.getRequestDispatcher("WEB-INF/views/login.jsp").forward(req, resp);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controller/LoginController.java]:
```suggestion
            req.getRequestDispatcher("/WEB-INF/views/login.jsp").forward(req, resp);
```

File: src/main/java/com/internet/shop/web/filter/AuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        if (userId == null || userService.get(userId) == null) {
+            resp.sendRedirect(req.getContextPath() + "/login");
+            return;
+        }
+        chain.doFilter(req, resp);
+    }
+
+    @Override
+    public void destroy() {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/web/filter/AuthenticationFilter.java]: redundant empty line

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/com/internet/shop/controller/LoginController.java (hunk 1 above)
Comment (Rommelua) [reply src/main/java/com/internet/shop/controller/LoginController.java]: fixed

File: src/main/java/com/internet/shop/web/filter/AuthenticationFilter.java (hunk 2 above)
Comment (Rommelua) [reply src/main/java/com/internet/shop/web/filter/AuthenticationFilter.java]: fixed

**Pull request merged**